provider/kubernetes/kubernetes.go

```diff
@@ -68,10 +68,13 @@ func (p *Provider) newK8sClient(ctx context.Context, ingressLabelSelector string
 	}
 
 	var cl *clientImpl
-	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" && os.Getenv("KUBERNETES_SERVICE_PORT") != "" {
-		log.FromContext(ctx).Infof("Creating in-cluster Provider client%s", withEndpoint)
+	switch {
+	case os.Getenv("KUBERNETES_SERVICE_HOST") != "" && os.Getenv("KUBERNETES_SERVICE_PORT") != "":
 		cl, err = newInClusterClient(p.Endpoint)
-	} else {
+	case os.Getenv("KUBECONFIG") != "":
+		log.FromContext(ctx).Infof("Creating cluster-external Provider client from KUBECONFIMG %s", os.Getenv("KUBECONFIG"))
+		cl, err = newExternalClusterClientFromFile(os.Getenv("KUBECONFIG"))
+	default:
 		log.FromContext(ctx).Infof("Creating cluster-external Provider client%s", withEndpoint)
```

Comment:
```suggestion
		log.FromContext(ctx).Infof("Creating cluster-external Provider client: %s", withEndpoint)
```

Reply:
in fact the `withEndpoint` is filled before with 
```golang
withEndpoint := ""
if p.Endpoint != "" {
    withEndpoint = fmt.Sprintf(" with endpoint %v", p.Endpoint)
}
```